jgrapht-core/src/main/java/org/jgrapht/alg/scoring/PageRank.java

```diff
@@ -0,0 +1,388 @@
+/* ==========================================
+ * JGraphT : a free Java graph-theory library
+ * ==========================================
+ *
+ * Project Info:  http://jgrapht.sourceforge.net/
+ * Project Creator:  Barak Naveh (http://sourceforge.net/users/barak_naveh)
+ *
+ * (C) Copyright 2003-2008, by Barak Naveh and Contributors.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+/* -------------------------
+* PageRank.java
+* -------------------------
+* (C) Copyright 2016, by Dimitrios Michail and Contributors.
+*
+* Original Author:  Dimitrios Michail
+* Contributor(s):   - 
+*
+* $Id$
+*
+* Changes
+* -------
+* 14-Aug-2016: Initial revision (DM);
+*
+*/
+package org.jgrapht.alg.scoring;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.interfaces.VertexScoringAlgorithm;
+
+/**
+ * PageRank implementation.
+ * 
+ * <p>
+ * The calculation is an iterative process which stops after a given number of
+ * iterations or if the PageRank values differences are less than a predefined
+ * value. This implementation uses the variant which divides one minus the
+ * damping factor by the number of nodes.
+ * </p>
+ * 
+ * <p>
+ * For an explanation of PageRank see this
+ * <a href="http://infolab.stanford.edu/~backrub/google.html">page</a> or the
+ * article: Sergey Brin and Larry Page: The Anatomy of a Large-Scale
```

Comment:
This is a rather straightforward implementation of the PageRank method based on the definition. I did not use any particular source for implementing it. 

However, I had a look at the source code of four different implementations 
 - Boost (BGL),
 - JUNG
 - igraph (their initial implementation as now they are using a much more sophisticated implementation using computer algebra packages, see [prpack](https://github.com/dgleich/prpack))
 - Gephy 

I update the javadoc to describe a little bit more the variant, to include the complexity and a link to the wikipedia page.